frontend/src/metabase-lib/Question.ts

```diff
@@ -1249,6 +1249,21 @@ class Question {
     return getIn(this, ["_card", "moderation_reviews"]) || [];
   }
 
+  /**
+   * We can only "explore results" (i.e. create new questions based on this one)
+   * when question is a native query, which is saved, has no parameters
+   * and satisfies other conditionals below.
+   */
+  canExploreResults() {
+    return (
+      this.isNative() &&
+      this.isSaved() &&
+      this.parameters().length === 0 &&
+      this.query().canNest() &&
+      !this.query().readOnly() // originally "canRunAdhocQuery"
+    );
+  }
+
```

Comment:
I was a bit hesitant to enlarge this god-object even further but it's still better than keeping this logic in essentially a view.

A nice consequence of that is that we could easily test this behavior with just a few unit tests as it is a pure function, which doesn't require any mocking or integration. 

I've seen that there's no tests for the `Question` now. Should I add them?

Reply:
This is a (very) legacy file https://github.com/metabase/metabase/blob/bf45689cd60bf337bc93bc7e23abf98f4a481a91/frontend/test/metabase-lib/lib/Question.unit.spec.js#L324 but that's what we have right now. We should add tests for this change.

Reply:
Btw, for new things we don't separate test files from the implementation files